whelktool/src/main/groovy/datatool/WhelkTool.groovy

```diff
@@ -440,6 +457,20 @@ class WhelkTool {
         }
     }
 
+    private void doSaveNew(DocumentItem item) {
+        Document doc = item.doc
+        doc.deepReplaceId(Document.BASE_URI.toString() + IdGenerator.generate())
+        doc.setControlNumber(doc.getShortId())
+        doc.setGenerationDate(new Date())
+        doc.setGenerationProcess(scriptJobUri)
+        if (!dryRun) {
+            Storage component = skipIndex ? whelk.storage : whelk
+            if (!component.createDocument(doc, changedIn, scriptJobUri,
+                    LegacyIntegrationTools.determineLegacyCollection(doc, whelk.getJsonld()), false))
+                throw new WhelkException("Failed to save a new document. See general whelk log for details.")
+        }
```

Comment:
This won't work because PostgreSQLComponent doesn't return false on errors but throws